aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::natives::{
+    aggregator_natives::helpers_v2::{
+        aggregator_snapshot_string_info, aggregator_snapshot_u128_info,
+        aggregator_snapshot_u64_info, string_value_info,
+    },
+    AccountAddress,
+};
+use aptos_gas_schedule::gas_params::natives::aptos_framework::*;
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError,
+    SafeNativeResult,
+};
+use move_core_types::value::{MoveStructLayout, MoveTypeLayout};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Struct, StructRef, Value},
+};
+use smallvec::{smallvec, SmallVec};
+use std::{collections::VecDeque, ops::Deref};
+
+/// The generic type supplied supplied to aggregator snapshots is not supported
+pub const EUNSUPPORTED_AGGREGATOR_SNAPSHOT_TYPE: u64 = 0x02_0005;
```

Comment:
Why are we starting from 5, what are 1-4?

Reply:
They are used for overflows, underflows, etc.

Reply:
But these are for aggregators?

Reply:
Btw 0x2 is used for out of range I think, we can maybe use other status here, e.g. 0x3 which is for invalid state. See `error.move`.

Reply:
Okay. I will switch to 0x3.
Aggregators_v2 native functions also will be eventually included in the same file and they were already using 1-4 errors, so I didn't use them.

Reply:
I see the conversation now :)